### PR TITLE
implement session callbacks (and echo plugin)

### PR DESCRIPTION
**Pull request by ShPelles**

### CODE REVIEW

File: Plugins/Echo/Echo.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        public PluginOutput Execute(string input, string session, ICallbacks callbacks)
+        {
+            if (input == "")
+            {
+                callbacks.StartSession();
+                return new PluginOutput("Echo started. Enter 'Exit' to stop.", null);
+            }
+            else if (input.ToLower() == "exit")
+            {
+                callbacks.StartSession();
```
Comment (YacovGold) [inline Plugins/Echo/Echo.cs]: StopSession?

File: Infrastructure/Callbacks.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+using System;
+using BasePlugin.Interfaces;
+
+namespace Infrastructure
+{
+    public class Callbacks : ICallbacks
+    {
+        // public IPlugin Plugin { get; set; }
+        // public bool IsActive { get; private set; }
```
Comment (YacovGold) [inline Infrastructure/Callbacks.cs]: Why comments?

File: Plugins/Echo/Echo.cs (hunk 1 above)
Comment (ShPelles) [reply Plugins/Echo/Echo.cs]: Sorry. My mistake. I only had half an hour...

File: Infrastructure/Callbacks.cs (hunk 2 above)
Comment (ShPelles) [reply Infrastructure/Callbacks.cs]: You're right. I'm removing it.

**Pull request merged**